11_FFTBloom/app_resources/kernel_spectrum_normalize.hlsl

```diff
@@ -0,0 +1,124 @@
+#include "common.hlsl"
+#include "nbl/builtin/hlsl/colorspace/encodeCIEXYZ.hlsl"
+
+/*
+ * Remember we have these defines:
+ * _NBL_HLSL_WORKGROUP_SIZE_
+ * ELEMENTS_PER_THREAD
+ * KERNEL_SCALE
+ * scalar_t
+ * FORMAT
+*/
+
+#define IMAGE_SIDE_LENGTH (_NBL_HLSL_WORKGROUP_SIZE_ * ELEMENTS_PER_THREAD)
```

Comment:
again supply from a Consteval struct

Reply:
maybve you want to make a 
```cpp
namespace nbl::hlsl::workgroup::fft
{
template<uint16_t _WorkgroupSizeLog2, uint16_t _ElementsPerThread> NBL_STRUCT_REQUIRES(mpl::is_pot_v<_ElementsPerThread>)
struct ConstevalParams
{
   NBL_CONSTEXPR_STATIC_INLINE uint16_t WorkgroupSizeLog2 = _WorkgroupSizeLog2;
   NBL_CONSTEXPR_STATIC_INLINE uint16_t _ElementsPerThread = _ElementsPerThread;

   NBL_CONSTEXPR_STATIC_INLINE uint32_t TotalSize = uint32_t(_ElementsPerThread)<<WorkgroupSizeLog2 ;
};
}
```
then your FFT struct could be partially specialized as
```cpp
template<typename consteval_params_t> // primary

template<uint16_t WorkgroupSizeLog2> // partial
struct FFT<ConstevalParams<WorkgroupSizeLog2,2> >

template<uint16_t WorkgroupSizeLog2, uint16_t N> // partial
struct FFT<ConstevalParams<WorkgroupSizeLog2,N> >
```

p.S. I don't actually remeber the clause for the `NBL_STRUCT_REQUIRES`